### PR TITLE
Spec changes specialize generic method

### PR DESCRIPTION
**Pull request by mppf**
This is a continuation of PR #5057.

*  Add a new test case for specific-instantiation 
*  Spec changes describing overload-specific method

reviewed by @vasslitvinov - thanks!

### CODE REVIEW

File: spec/Classes.tex [hunk 1]
```diff
@@ -214,17 +214,25 @@ \subsection{Class Methods}
...
 Methods defined outside of such scopes are known as \emph{secondary
 methods} and must have a \sntx{type-binding} (otherwise, they would
 simply be standalone functions rather than methods).  Note that
 secondary methods can be defined not only for classes, records, and
 unions, but also for any other type (e.g., integers, reals, strings).
 
+Secondary methods can be declared with a type expression instead of a
+type identifier. In particular, if the \sntx{type-binding} includes a
+parenthesized expression, the compiler will evaluate that expression to
+find the receiver type for the method. See also
```
Comment (vasslitvinov) [inline spec/Classes.tex]: "includes" -> "is" a parenthesized expression

Before "See also", I would add a sentence like "If so, the method applies only to receivers of this type."

File: spec/Generics.tex [hunk 2]
```diff
@@ -868,6 +868,81 @@ \section{User-Defined Compiler Diagnostics}
...
 
+\section{Declaring Specific Instantiations}
+\label{Declaring_Specific_Instantiations}
+\index{specific instantiations}
+\index{generic specialization}
+
+The Chapel language facility supports three mechanisms for creating generic
+functions along with concrete functions that work with specific types.
+One can use these mechanisms to create specialized versions of generic
+functions.
```
Comment (vasslitvinov) [inline spec/Generics.tex]: ... three mechanisms for *using* generic functions ...

I"d drop "that work with specific types".

"to create specialized versions" -- I think you mean "to choose between" generic and/or concrete functions. The last sentence above does not seem right to me as-is. Maybe blend it with the first sentence in the paragraph by using "choose between" alongside with "use" verbs somehow.

I suggest replacing, throughout this section, "specialization" with "instantiation" to be consistent with other parts of the spec.

**Comment by mppf** [comment]
@vasslitvinov - thanks for your feedback. Do you think "specific instantiation" is a good term for this ? I think C++ would call it "template specialization".

**Comment by mppf** [comment]
@vasslitvinov - I've adjusted based upon your feedback. Now the main thing I'm not quite happy with is the title for the new section in Generics.tex. Let me know if you have any suggestions there.

**Comment by mppf** [comment]
@vasslitvinov - I think this is good to go now - do you agree?

**Comment by vasslitvinov** [comment]
@mppf yes, sounds good. thank you!

**Pull request merged**